### ISSUE TITLE
File permissions change on Linux after reading the file on Windows

### ISSUE THREAD

**Issue by herrmanthegerman** [Microsoft/WSL#3845]
My _ver_ is  10.0.17134.523

**tl;dr:** I'm seeing files on my external NTFS-formatted hard drive that I cannot access through Linux (wsl Ubuntu) until I access/read them once through Windows which changes permissions on the Linux side.

I have an external hard drive (NTFS formatted) that is mounted on WSL Ubuntu using  `mount -t drvfs d: /mnt/d -o uid=1000,gid=1000`

Some files on the drive cannot be accessed by WSL due to lack of permissions, e.g.,

```
daddy@Dad:/mnt/d/family$ stat Imperial\ March.pdf
  File: 'Imperial March.pdf'
  Size: 197675          Blocks: 392        IO Block: 4096   regular file
Device: fh/15d  Inode: 281474976919536  Links: 1
Access: (0000/----------)  Uid: ( 1000/   daddy)   Gid: ( 1000/   daddy)
Access: 2019-01-27 12:22:12.320206300 +0100
Modify: 2019-01-26 00:13:51.845373300 +0100
Change: 2019-01-26 00:13:51.845373300 +0100
```

However, on the Windows side, my user account that is running WSL has full access to the file. 

Now if I access this file on Windows at least once, e.g., using a simple "dir" on it or (a more advanced) powershell command:

```
PS D:\family> get-acl '.\Imperial March.pdf'
    Directory: D:\family
Path               Owner                        Access
----               -----                        ------
Imperial March.pdf VORDEFINIERT\Administratoren Jeder Allow  FullControl
```

... the permissions on the Linux side change:

```
daddy@Dad-Aero15:/mnt/d/family$ stat Imperial\ March.pdf
  File: 'Imperial March.pdf'
  Size: 197675          Blocks: 392        IO Block: 4096   regular file
Device: fh/15d  Inode: 281474976919536  Links: 1
Access: (0777/-rwxrwxrwx)  Uid: ( 1000/   daddy)   Gid: ( 1000/   daddy)
Access: 2019-01-27 12:22:12.320206300 +0100
Modify: 2019-01-26 00:13:51.845373300 +0100
Change: 2019-01-26 00:13:51.845373300 +0100
 Birth: -
```

The permissions stick until the hard drive is unmounted and removed from Windows. 

As you may have noticed, the file owner is "VORDEFINIERT\Administratoren" and not my user account running WSL.

Nevertheless, I expected to have full access to that file on the Linux side immediately after mounting the drive.

What is going on here? Thanks.

**Comment by herrmanthegerman** [comment]
As a workaround, I changed the ownership of the top folder "family" (properties -> security -> advanced ->  owner -> change) to the account running WSL and checked the box that says "Replace all child object permissions entries with inheritable permission entries from this object" before applying.

**Comment by herrmanthegerman** [comment]
For a similar observation that might be related, see this report:
https://superuser.com/questions/1352207/windows-wsl-ubuntu-sees-wrong-permissions-on-files-in-mounted-disk

**Comment by lewisdonofrio** [comment]
I always copy my files like if I am doing a java war file deploy, I download the file from firefox win64, it drops into "c:\Users\username\Downloads" then I cp the file from ubuntu/bash as cp /mnt/c/Users/username/Downloads/hellowworld.war.... just saying stay away from any dos4unix issues (https://kb.iu.edu/d/acux) and vise versa, if I want to edit a png/tiff I cp the file again same as outlined then I run gimp in ubuntu make the changes as needed and save it, then cp the file out of ubuntu homespace to windows homespace.  this logic when followed works all the time, been working fine for me for over two years now....thank you for all you folks do with WSL it's a godsend!

**Comment by SvenGroot** [comment]
@herrmanthegerman I'm unable to reproduce this issue on Windows 10 version 1809 (10.0.17763.316).

Any chance you can upgrade and try again?

**Comment by herrmanthegerman** [comment]
@SvenGroot Thanks, I'm still waiting  ["until the feature update is offered to my device automatically"](https://support.microsoft.com/en-in/help/4464619/windows-10-update-history). As soon as this happens, I'll try again.

**Comment by mrother** [comment]
Hi,

I'm on Windows 10 version 2004 (10.0.19041.1) and having the same problem:

External HDD with NTFS, after reboot the permissions goes crazy. If I open the folder on Windows, the permissions are restored, but not recursively. I just gave up using external HDD for now.

**Comment by LearnedVector** [comment]
bump! i'm also having this issue. any fix?

**Comment by YunPengYue** [comment]
> As a workaround, I changed the ownership of the top folder "family" (properties -> security -> advanced -> owner -> change) to the account running WSL and checked the box that says "Replace all child object permissions entries with inheritable permission entries from this object" before applying.

change to the account running WSL? but i can't find root or my wsl other user name,
